midterm-2023-11-06/code/5215-berat-useini/script.ts

```diff
@@ -0,0 +1,191 @@
+type RockBand = {
+    ID: number;
+    name: string;
+    formed: number;
+    location: string;
+    country: string;
+    members: string;
+    albums: number;
+};
+
+interface albums{ 
+name: string;
+
+}
+
+document.addEventListener("DOMContentLoaded", siteCode);
+let rockband : RockBand[] = [];
+async function siteCode() {
+    const data = await loadData();
+    rockband = data.metalBands;
+
+    fillLocation(rockband)
+
+    const sortButton = document.getElementById("sort-id")!
+    sortButton.addEventListener("click", sortById)
+
+
+    const sortName = document.getElementById("sort-name")!
+    sortName.addEventListener("click", sortByName)
+
+    const applyName = document.getElementById("name-filter")
+    applyName?.addEventListener("input", applyNameFilter)
+
+ 
+
+    const applyButton = document.getElementById("apply-filter")!
+    applyButton.addEventListener("click", applyFilter)
+    populateTable(rockband)
+
+
+}
+
+// Function to populate the HTML table with rock band data
+
+const fillLocation = (rockband: RockBand[]) => {
+    const filter = document.getElementById("country-filter")!;
```

Comment:
As a consequence of the interfaces not reflecting the data, it's not obvious that the data does **not** have a country property, i.e. the set will only hold the value `undefined`.

Otherwise, the format of the method is correct +5